packages/hash/frontend/src/pages/[account-slug]/types/entity-type/edit-bar.tsx

```diff
@@ -0,0 +1,110 @@
+import { faSmile } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@hashintel/hash-design-system/fontawesome-icon";
+import { Box, Container, Stack, Typography } from "@mui/material";
+import { ReactNode } from "react";
+import { PencilSimpleLine } from "../../../../shared/icons/svg";
+import { Button, ButtonProps } from "../../../../shared/ui/button";
+
+const EditBarContents = ({
+  icon,
+  title,
+  label,
+  discardButtonProps,
+  confirmButtonProps,
+}: {
+  icon: ReactNode;
+  title: ReactNode;
+  label: ReactNode;
+  discardButtonProps: ButtonProps;
+  confirmButtonProps: ButtonProps;
+}) => (
```

Comment:
I remember us chatting about potential removal of `FunctionComponent` in code, but if we go for it, we should probably do so in some organised way. I'd stick with `FunctionComponent` for consistency for now, especially for exportable components. Defining `type FooBarProps` separately can also contribute to overall consistency. I'm not strongly opinionated about any of the styles, just suggesting to keep code rhythm as consistent as possible.